core-plugins/src/main/java/co/cask/hydrator/plugin/batch/sink/KVTableSink.java

```diff
@@ -91,9 +119,9 @@ public KVTableSink(KVTableConfig kvTableConfig) {
   public void transform(StructuredRecord input, Emitter<KeyValue<byte[], byte[]>> emitter) throws Exception {
     Object key = input.get(kvTableConfig.keyField);
     Preconditions.checkArgument(key != null, "Key cannot be null.");
-    byte[] keyBytes = key instanceof ByteBuffer ? Bytes.toBytes((ByteBuffer) key) : (byte[]) key;
+    byte[] keyBytes = key instanceof String ? Bytes.toBytes((String) key) : (byte[]) key;
```

Comment:
can't get rid of the ByteBuffer logic. The previous logic should be applied if the schema says the field's type is bytes.  If the schema says the field's type is string, then you can cast it directly to a String and just do Bytes.toBytes() on it.